src/main/python/util.py

```diff
@@ -57,6 +56,55 @@ def config_input_check(text: str, available: int, res: Enum):
             return res.INT_ERROR
 
 
+def build_docker_container(detached=True, container_path='samgomena/deepshare_worker', version='latest', memory=None, cpus=None):
+    """creates and runs a docker container from:
+    https://hub.docker.com/u/samgomena/repository/docker/samgomena/deepshare_worker
+    Returns the container ID if successful, False if not
+    """
+    docker_client = check_and_instantiate_docker_client()
```

Comment:
I see there's a `_docker_initial_check` which has called `check_and_instantiate_docker_client()`, then we might be we can assume user have it when they are on `add_resource` page (unless they uninstall it when they are on the page and before submit it, ha)

Reply:
so the reason that I did it this way is that I couldn't figure out a better way to check if the user has docker installed on their system for `docker-py` other than actually instantiating the docker client itself (it throws an exception otherwise).  So when I initially call it upon the add_recource page, we don't care about the return value because we just want to know _if_ we can make a docker client.  Later when we call it we actually want the return value

Reply:
@lixphilosophy is this ok or do you still think I should change it?

Reply:
I think we can just take this right now, since I don't have any better idea about how we can deal with this. It just once extra call, won't bother too much.